### ISSUE TITLE
Uwsgi pmda does not update metric values when server stops

### ISSUE THREAD

**Issue by sfeifer** [performancecopilot/pcp#1964]
When I start the uwsgi server, the metric values update correctly. Upon stopping the server though, the metric values persist. For example, I start the server with 1 worker and uwsgi.summary.total_workers gets updated to have a value of 1. When I stop the server, uwsgi.summary.total_workers keeps its value of 1, even though there is no longer a uwsgi server running.

**Comment by natoscott** [comment]
@jainnikhil30 I took a quick look and I think this is a partial fix, but @sfeifer mentioned it doesn't completely solve the problem in grafana-pcp...

```
git diff
src/pmdas/uwsgi/pmdauwsgi.python
--- /tmp/git-blob-EJVmCw/pmdauwsgi.python       2024-04-23 10:27:25.584880409 +1000
+++ src/pmdas/uwsgi/pmdauwsgi.python    2024-04-19 08:17:11.134455391 +1000
@@ -185,7 +185,7 @@ class UwsgiPMDA(PMDA):
             #self.log("refresh_all stats:", str(stats))
         except Exception:
             self.get_failed = True
-            return
+            stats = {}
 
         avg_response_time_msec = 0
         summary = UWSGISUMMARY()
@@ -213,7 +213,8 @@ class UwsgiPMDA(PMDA):
                 summary.total_pause_worker_count += 1
             avg_response_time_msec += worker.avg_rt
             self.workers[str(worker_id)] = worker
-        summary.avg_response_time_msec = avg_response_time_msec/summary.total_workers
+        if summary.total_workers > 0:
+            summary.avg_response_time_msec = avg_response_time_msec / summary.total_workers
         self.summary = summary
 
     def read_config(self):
```

**Comment by natoscott** [comment]
Fixed in pcp-6.2.2
